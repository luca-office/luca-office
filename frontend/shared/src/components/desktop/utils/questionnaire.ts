import {ScenarioQuestionnairesQuery_scenarioQuestionnaires} from "../../../graphql/generated/ScenarioQuestionnairesQuery"

export const questionnaireTypeForQuestionId = (
  questionId: UUID,
  questionnaireId: UUID,
  scenarioQuestionnaires: ScenarioQuestionnairesQuery_scenarioQuestionnaires[]
) =>
  scenarioQuestionnaires
    .filter(scenarioQuestionnaire => scenarioQuestionnaire.questionnaireId === questionnaireId)
    .map(scenarioQuestionnaires => scenarioQuestionnaires.questionnaire)
    .flatMap(questionnaire => questionnaire.questions)
    .find(question => question.id === questionId)?.questionType
