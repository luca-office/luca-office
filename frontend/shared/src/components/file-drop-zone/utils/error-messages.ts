import {LucaI18nLangKey} from "../../../translations"

export enum FileUploadError {
  FileTooLarge,
  InvalidFileType,
  TooManyFiles
}

export const getLanguageKeyForFileUploadError = (error: FileUploadError): LucaI18nLangKey => {
  switch (error) {
    case FileUploadError.FileTooLarge:
      return "files_and_directories__upload_modal_file_too_large"
    case FileUploadError.InvalidFileType:
      return "files_and_directories__upload_modal_file_type_invalid"
    case FileUploadError.TooManyFiles:
      return "files_and_directories__upload_modal_no_single_file"
  }
}

export type FileErrorMessage = "file-too-large" | "file-too-small" | "too-many-files" | "file-invalid-type"

export const mapDropzoneRejectionToFileUploadError = (errorMessage: FileErrorMessage) => {
  switch (errorMessage) {
    case "file-invalid-type":
      return FileUploadError.InvalidFileType
    case "file-too-large":
      return FileUploadError.FileTooLarge
    case "too-many-files":
      return FileUploadError.TooManyFiles
    default:
      return FileUploadError.InvalidFileType
  }
}
