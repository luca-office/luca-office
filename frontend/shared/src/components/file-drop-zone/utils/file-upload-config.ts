export const maximumFileSizeInMegaByte = 100
export const maximumFileSizeInBytes = maximumFileSizeInMegaByte * Math.pow(10, 6)
