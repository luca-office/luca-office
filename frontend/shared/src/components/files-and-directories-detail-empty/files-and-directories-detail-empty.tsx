import {css} from "@emotion/react"
import * as React from "react"
import {HeadingLevel, IconName} from "../../enums"
import {fontColorLight, spacingSmall} from "../../styles"
import {LucaTFunction} from "../../translations"
import {FilesAndDirectoriesDetailView, Heading} from ".."

interface Props {
  readonly t: LucaTFunction
}

export const FilesAndDirectoriesDetailEmpty: React.FC<Props> = ({t}) => (
  <FilesAndDirectoriesDetailView
    headerIcon={IconName.Folder}
    customCardContentStyles={styles.cardContent}
    headerTitle={t("files_and_directories__detail__empty__title")}>
    <div css={styles.content}>
      <Heading level={HeadingLevel.h3}>{t("files_and_directories__detail__empty__heading")}</Heading>
      <Heading level={HeadingLevel.h3} customStyles={styles.text}>
        {t("files_and_directories__detail__empty__text")}
      </Heading>
    </div>
  </FilesAndDirectoriesDetailView>
)

const styles = {
  content: css({
    flex: "1 1 auto",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center"
  }),
  text: css({
    marginTop: spacingSmall,
    color: fontColorLight
  }),
  cardContent: css({
    overflow: "auto"
  })
}
