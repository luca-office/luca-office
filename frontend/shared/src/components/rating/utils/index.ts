export * from "./coding-criterion"
export * from "./coding-dimension"
export * from "./coding-item"
export * from "./question"
export * from "./score"
export * from "./scoring-type"
