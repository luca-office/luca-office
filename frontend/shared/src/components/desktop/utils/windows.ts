import findLastIndex from "lodash-es/findLastIndex"
import {OfficeTool, OfficeWindowType} from "../../../enums"
import {Option} from "../../../utils/option"

export const findVisibleWindows = (openedWindows: OfficeWindowType[], minimizedWindows: OfficeWindowType[]) => {
  const visibleWindows = openedWindows.filter(window => !minimizedWindows.includes(window))

  return Option.of(findLastIndex(visibleWindows, isFullScreenWindow))
    .map(lastFullScreenToolIndex => visibleWindows.slice(lastFullScreenToolIndex))
    .getOrElse(visibleWindows)
}

const isFullScreenWindow = (window: OfficeWindowType) => {
  switch (window) {
    case OfficeTool.Notes:
    case OfficeTool.Calculator:
    case OfficeTool.Chat:
      return false
    default:
      return true
  }
}
