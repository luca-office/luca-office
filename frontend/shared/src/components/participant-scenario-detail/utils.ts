import {FreetextAnswerFragment} from "../../graphql/generated/FreetextAnswerFragment"
import {QuestionType} from "../../graphql/generated/globalTypes"
import {Questionnaire, RuntimeSurveyResults} from "../../models"
import {ParticipantQuestionAnswer, QuestionnaireParticipantResults} from "../questionnaire"

export const runtimeSurveyResultsToQuestionnaireParticipantResults = (
  invitationId: UUID,
  questionnaire: Questionnaire,
  runtimeSurveyResults: RuntimeSurveyResults,
  freeTextAnswers: FreetextAnswerFragment[]
): QuestionnaireParticipantResults => {
  const answers = runtimeSurveyResults.questionResults.reduce<Record<UUID, ParticipantQuestionAnswer>>(
    (answersAccumulator, result) => {
      const question = questionnaire.questions.find(question => question.id === result.questionId)
      const participantResult = result.participantResults.find(results => results.invitationId === invitationId)
      const freeTextAnswer = freeTextAnswers.find(
        answer => answer.questionId === result.questionId && answer.surveyInvitationId === invitationId
      )?.text

      switch (question?.questionType) {
        case QuestionType.SingleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.SingleChoice,
              questionId: question.id,
              selectedAnswerId: participantResult?.selectedAnswerIds[0] ?? "",
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.MultipleChoice: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.MultipleChoice,
              questionId: question.id,
              selectedAnswerIds: participantResult?.selectedAnswerIds ?? [],
              isFreetextAnswerSelected: participantResult?.wasFreetextAnswerSelected ?? false,
              freeTextAnswer: freeTextAnswer
            }
          }
        }

        case QuestionType.FreeText: {
          return {
            ...answersAccumulator,
            [question.id]: {
              questionType: QuestionType.FreeText,
              questionId: question.id,
              freeTextAnswer: freeTextAnswer ?? ""
            }
          }
        }

        default:
          return answersAccumulator
      }
    },
    {}
  )

  return {questionnaireId: runtimeSurveyResults.questionnaireId, answers}
}
