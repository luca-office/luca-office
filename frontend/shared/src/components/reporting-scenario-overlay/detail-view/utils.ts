import {TableEntity} from "../../../components/rating/models"
import {CodingDimension, CodingItem, CodingItemResultByItemId} from "../../../models"

export const itemToTableEntity = (
  item: CodingItem,
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: item.id,
  title: item.title,
  position: item.position,
  score: codingItemResultsByItem[item.id].score,
  maxScore: codingItemResultsByItem[item.id].maximumScore,
  averageScore: codingItemResultsByItem[item.id].averageScore,
  rated: true
})

export const dimensionToTableEntity = (
  dimension: CodingDimension,
  items: CodingItem[],
  codingItemResultsByItem: CodingItemResultByItemId
): TableEntity => ({
  id: dimension.id,
  title: dimension.title,
  position: dimension.position,
  score: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].score, 0),
  maxScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].maximumScore, 0),
  averageScore: items.reduce((acc, item) => acc + codingItemResultsByItem[item.id].averageScore, 0),
  rated: true
})
