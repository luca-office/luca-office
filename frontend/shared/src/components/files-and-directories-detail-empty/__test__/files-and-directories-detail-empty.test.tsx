// importing from direct file because of issues of babel loader and spyOn
import * as React from "react"
import {create} from "react-test-renderer"
import {fakeTranslate} from "../../../../tests/utils/translate-mock"
import {FilesAndDirectoriesDetailEmpty} from "../files-and-directories-detail-empty"

describe("Empty Detail View", () => {
  it("renders correctly", async () => {
    const component = create(<FilesAndDirectoriesDetailEmpty t={fakeTranslate} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
