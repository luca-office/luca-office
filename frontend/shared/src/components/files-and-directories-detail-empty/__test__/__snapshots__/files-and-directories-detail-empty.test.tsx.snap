// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Detail View renders correctly 1`] = `
<div
  className="css-1lb0na6-Card"
>
  <div
    className="card-header css-diisqk-CardHeader"
  >
    <div
      className="css-11jhu3e-FilesAndDirectoriesDetailView"
    >
      <div
        className="css-1lrzij5-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1i6jmor-Heading"
      >
        files_and_directories__detail__empty__title
      </div>
    </div>
    <div
      className="css-1400h4j-FilesAndDirectoriesDetailView"
    />
  </div>
  <div
    className="card-content css-1cbxobb-CardContent"
  >
    <div
      className="css-kv0ku-FilesAndDirectoriesDetailEmpty"
    >
      <div
        className="css-1f6p2vj-Heading"
      >
        files_and_directories__detail__empty__heading
      </div>
      <div
        className="css-109orsd-Heading"
      >
        files_and_directories__detail__empty__text
      </div>
    </div>
  </div>
</div>
`;
